internal/controller/controllers/infraenv_controller.go

```diff
@@ -502,7 +500,14 @@ func (r *InfraEnvReconciler) createInfraEnv(ctx context.Context, log logrus.Fiel
 	if cluster != nil {
 		clusterID = cluster.ID
 	}
-	createParams := CreateInfraEnvParams(infraEnv, r.getImageType(ctx, infraEnv), pullSecret, clusterID, osImageVersion)
+	imageType := r.getImageType(log, infraEnv)
+	infraEnv.Spec.ImageType = imageType
+	createParams := CreateInfraEnvParams(infraEnv, imageType, pullSecret, clusterID, osImageVersion)
+	if err = r.Update(ctx, infraEnv); err != nil {
+		err = errors.Wrapf(err, "failed to update InfraEnv %s with ImageType %s", infraEnv.Name, imageType)
+		log.WithError(err)
+		return nil, err
+	}
```

Comment:
I don't think it's preferable to update the InfraEnv's spec. The spec is there for the user to spec-ify their own configs and I don't think our controller should set this field, even if it was empty.

Reply:
So leave it empty? Even if we updated the image type to some default value?

Reply:
Yes, if you want, the status can have what we set it to. The default value is expected normally as the user couldn't request the type before